addon/components/imgix-image.js

```diff
@@ -1,9 +1,203 @@
-import Ember from 'ember';
-import ResizeMixin from 'ember-resize-mixin/main';
-import layout from '../templates/components/imgix-image';
-import ImgixPathBehavior from '../mixins/imgix-path-behavior';
-
-export default Ember.Component.extend(ResizeMixin, ImgixPathBehavior, {
-  layout: layout,
-  classNames: ['imgix-image-wrap']
+import Component from '@ember/component';
+import { computed, get, set } from '@ember/object';
+import ResizeAware from 'ember-resize-aware/mixins/resize-aware';
+import { merge } from '@ember/polyfills';
+import { tryInvoke } from '@ember/utils';
+import config from 'ember-get-config';
+import EmberError from '@ember/error';
+import ImgixClient from 'imgix-core-js';
+import { debounce } from '@ember/runloop';
+
+export default Component.extend(ResizeAware, {
+  tagName: 'img',
+  classNames: 'imgix-image',
+  attributeBindings: ['src', 'crossorigin', 'alt'],
+
+  path: null, // The path to your image
+  aspectRatio: null,
+  crop: 'faces',
+  fit: 'crop',
+  pixelStep: 1,
+  onLoad: null,
+  onError: null,
+  crossorigin: 'anonymous',
+  alt: '', // image alt
+  options: {}, // arbitrary imgix options
+
+  width: null, // passed in
+  height: null, // passed in
+
+  _width: null,
+  _height: null,
+  _dpr: null,
+
+  debounceRate: 400,
+
+  didResize(width, height) {
+    if (get(this, 'path')) {
+      const newWidth =
+        Math.ceil(
+          get(this, '_pathAsUrl.searchParams').get('w') ||
+            width / get(this, 'pixelStep')
+        ) * get(this, 'pixelStep');
+      const newHeight = Math.floor(
+        get(this, 'aspectRatio')
+          ? newWidth / get(this, 'aspectRatio')
+          : get(this, '_pathAsUrl.searchParams').get('h') || height
+      );
+      const newDpr = window.devicePixelRatio || 1;
```

Comment:
This value should be constrained--we don't want to send something like `?dpr=1.3333333`, which is possible if the user's browser zoom level is not set to 100%. We should trim this at a precision of 3, i.e. `?dpr=1.33`.

Reply:
I've pulled this out into an issue: #33